src/main/java/core/basesyntax/processing/RawDataProcessor.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.processing;
+
+import core.basesyntax.enums.Operation;
+import core.basesyntax.handler.TransactionHandler;
+import core.basesyntax.handler.impl.BalanceTransactionHandler;
+import core.basesyntax.handler.impl.PurchaseTransactionHandler;
+import core.basesyntax.handler.impl.ReturnTransactionHandler;
+import core.basesyntax.handler.impl.SupplyTransactionHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RawDataProcessor {
+    private final List<List<String>> rawData;
+    private final Map<String, TransactionHandler> transactionHandlerMap;
+
+    public RawDataProcessor(List<List<String>> rawData) {
+        this.rawData = rawData;
+        this.transactionHandlerMap = Map.of(
+                Operation.BALANCE.getCode(), new BalanceTransactionHandler(),
+                Operation.SUPPLY.getCode(), new SupplyTransactionHandler(),
+                Operation.PURCHASE.getCode(), new PurchaseTransactionHandler(),
+                Operation.RETURN.getCode(), new ReturnTransactionHandler()
+
+        );
```

Comment:
[where do we need ` this` ?](https://stackoverflow.com/a/2411283)
also, to make this class more flexible, let's create this map in Main and pass to constructor as an argument